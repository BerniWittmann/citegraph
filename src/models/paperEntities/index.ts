import Author from './author'
import Conference from './conference'
import Country from './country'
import Institution from './institution'
import Publication from './publication'
import Publisher from './publisher'
import Record from './record'
import PaperEntity, { PaperEntityTableColumn, PaperEntityFields } from './base'

export const entities = [
  Record,
  Author,
  Institution,
  Country,
  Publication,
  Publisher,
  Conference
]

export const entityKeysMap = {
  [Record.key]: Record,
  [Author.key]: Author,
  [Institution.key]: Institution,
  [Country.key]: Country,
  [Publication.key]: Publication,
  [Publisher.key]: Publisher,
  [Conference.key]: Conference
}

export const entityKeys: Array<string> = entities.map((entity) => entity.key)

export type PaperEntityTableColumn = PaperEntityTableColumn

export type PaperEntity = PaperEntity

export type PaperEntityFields = PaperEntityFields

export default {
  Author,
  Conference,
  Country,
  Institution,
  Publication,
  Publisher,
  Record
}
