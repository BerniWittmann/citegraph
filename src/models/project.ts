export default class Project {
  name: string;
  id?: number;

  constructor ({ name, id }: Project) {
    this.id = id
    this.name = name
  }
}
