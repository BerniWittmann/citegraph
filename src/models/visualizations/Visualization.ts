import Transformer from '@/transformers/Transformer'

export interface TimePeriod {
  from: number
  to: number
}

export interface VisualizationBaseFields {
  id?: string;
  progress?: number;
  name: string;
  timePeriods?: Array<TimePeriod>
}

export interface VisualizationBaseFieldsDTO {
  id?: string;
  progress?: number;
  name: string;
  key: string;
  // eslint-disable-next-line camelcase
  time_periods?: Array<TimePeriod>
}

export default abstract class Visualization implements VisualizationBaseFields {
  id?: string;
  progress?: number;
  name: string;
  timePeriods: Array<TimePeriod>
  abstract parameters: Object;
  abstract data?: Object;
  abstract readonly key: string;
  static readonly key: string;
  static readonly title: string;
  static readonly imageUrl: string;
  static readonly description: string;
  static readonly longDescription: string;
  static readonly transformer: typeof Transformer;

  protected constructor ({ id, progress, name, timePeriods }: VisualizationBaseFields) {
    this.id = id
    this.progress = progress
    this.name = name
    this.timePeriods = timePeriods || []
  }
}
