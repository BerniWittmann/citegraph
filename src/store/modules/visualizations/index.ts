import { Module } from 'vuex'
import { VisualizationsState } from './types'
import { RootState } from '@/store/types'
import { state } from './state'
import { actions } from './actions'
import { getters } from './getters'
import { mutations } from './mutations'

const namespaced: boolean = true

export const visualizations: Module<VisualizationsState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations
}
