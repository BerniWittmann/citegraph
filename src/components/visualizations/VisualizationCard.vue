<template>
  <v-card :hover="true"
          class="visualization-card"
          height="100%">
    <div class="d-flex flex-no-wrap justify-space-between">
      <v-avatar
        class="ma-3"
        size="125"
        tile
      >
        <v-img contain :src="imageUrl"/>
      </v-avatar>
      <div class="flex-grow-1 d-flex flex-column align-center">
        <v-card-title>{{ visualization.name }}</v-card-title>
        <v-card-text class="progress-information text-center">
          <span v-if="isLoading">{{ $t('visualizations.card.processing', { percent: progress }) }}</span>
          <span v-else>{{ $t('visualizations.card.ready') }}</span>
        </v-card-text>
        <v-divider style="width: 90%" class="mx-auto"></v-divider>
        <v-card-actions>
          <v-btn color="primary" text>
            {{ $t('visualizations.card.show') }}
          </v-btn>
          <v-btn color="primary" text>
            {{ $t('visualizations.card.edit') }}
          </v-btn>
        </v-card-actions>
      </div>
    </div>
    <v-progress-linear color="#F37F21" background-color="#FBBF35" background-opacity="0.5" v-show="isLoading" class="progress" :value="progress" :max="100"></v-progress-linear>
  </v-card>
</template>
<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import Visualization from '../../models/visualizations/Visualization'
import { visualizationsKeyMap } from '@/models/visualizations'

@Component
export default class VisualizationCard extends Vue {
  @Prop() readonly visualization!: Visualization

  get visualizationClass () {
    return visualizationsKeyMap[this.visualization.key]
  }

  get imageUrl (): string {
    return `${process.env.BASE_URL}${this.visualizationClass.imageUrl}`
  }

  get progress (): number {
    return this.visualization.progress ? this.visualization.progress * 100 : 0
  }

  get isLoading (): boolean {
    return !!this.visualization.progress && (this.progress < 100)
  }
}
</script>

<style scoped lang="scss">
.progress-information {
  color: #F37F21;
}
</style>
