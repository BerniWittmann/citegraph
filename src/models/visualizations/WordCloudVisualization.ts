import Visualization, { VisualizationBaseFields } from './Visualization'

export interface WordCloudVisualizationParameters {
  pruneLeastCitedPercentage: number
}

export interface WordCloudDataPoint {
  word: string
  weight: number
}
export interface WordCloudData {
  data: Array<WordCloudDataPoint>
}

export interface WordCloudVisualizationFields {
  data?: WordCloudData
}

export default class WordCloudVisualization extends Visualization implements WordCloudVisualizationFields {
  static key: string = 'wordcloud'
  key: string = 'wordcloud'
  static title: string = 'visualizations.information.word_cloud.title'
  static description: string = 'visualizations.information.word_cloud.description'
  static longDescription: string = 'visualizations.information.word_cloud.long_description'
  static imageUrl: string = 'visualization_images/word_cloud.jpg'
  parameters: WordCloudVisualizationParameters
  data?: WordCloudData

  constructor ({ id, name, progress, data }: VisualizationBaseFields & WordCloudVisualizationFields) {
    super({ id, name, progress })
    this.parameters = {
      pruneLeastCitedPercentage: 5
    }
    this.data = data
  }
}
