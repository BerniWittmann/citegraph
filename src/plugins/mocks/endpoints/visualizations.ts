import { AxiosRequestConfig } from 'axios'
import MockAdapter from 'axios-mock-adapter'
import db from '../db'
import Project from '@/models/project'
import Visualization from '@/models/visualizations/Visualization'

const projects = db.getCollection('projects')
const visualizations = db.getCollection('visualizations')

function getProjectId (config: AxiosRequestConfig): number {
  const urlParts: Array<string> = config.url!.split('/')
  return parseInt(urlParts[4])
}

function getVisualizationId (config: AxiosRequestConfig): string {
  const urlParts: Array<string> = config.url!.split('/')
  return urlParts[6]
}

export function setupMocks (mock: MockAdapter) {
  mock.onGet(/\/projects\/\d+\/visualizations$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    return [200, visualizations.find()]
  })

  mock.onGet(/\/projects\/\d+\/visualizations\/\d+$/).reply((config: AxiosRequestConfig) => {
    const project: Resultset<Project> = projects.findOne({ 'id': getProjectId(config) })
    if (!project) {
      return [404]
    }
    const visualization: Resultset<Visualization> = visualizations.findOne({ 'id': getVisualizationId(config) })
    if (!visualization) {
      return [404]
    }
    return [200, visualization]
  })
}
