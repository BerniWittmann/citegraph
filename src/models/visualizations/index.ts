import WordCloudVisualization, { WordCloudVisualizationFields } from './WordCloudVisualization'
import { VisualizationBaseFields } from '@/models/visualizations/Visualization'
import BarChartVisualization, { BarChartVisualizationFields } from '@/models/visualizations/BarChartVisualization'

export const visualizations = [WordCloudVisualization, BarChartVisualization]

const visualizationsKeyMap: Record<string, any> = {}
visualizations.forEach((visualization) => {
  visualizationsKeyMap[visualization.key] = visualization
})

export {
  visualizationsKeyMap
}

export type VisalizationFields = VisualizationBaseFields & (
  WordCloudVisualizationFields &
  BarChartVisualizationFields
)
