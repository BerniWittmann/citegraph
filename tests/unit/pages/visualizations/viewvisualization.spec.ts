import { shallowMount } from '@vue/test-utils'
import { i18n } from '../../setupPlugins'

import ViewVisualizationPage from '@/pages/visualizations/visualization/ViewVisualization.vue'
import Visualization from '@/models/visualizations/Visualization'
import { visualizations } from '@/models/visualizations'
import WordCloudVisualization from '@/models/visualizations/WordCloudVisualization'

describe('pages/visualizations/EditVisualization.vue', () => {
  function getWrapper (visualization?: Visualization) {
    const store = {
      getters: {
        'visualizations/hasCurrentVisualization': !!visualization,
        'visualizations/currentVisualization': visualization
      }
    }

    return shallowMount(ViewVisualizationPage, {
      i18n,
      mocks: {
        $route: {
          params: {
            projectId: 23,
            visualizationId: 1
          }
        },
        $store: store
      }
    })
  }

  visualizations.forEach(VisualizationClass => {
    describe('renders the page for ' + VisualizationClass.key, () => {
      const vis = new VisualizationClass({
        id: '1',
        name: 'My Visualization'
      })
      it('renders', () => {
        vis.data = {
          foo: 'bar'
        }
        vis.progress = 1
        const wrapper = getWrapper(vis)
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.contains('v-progress-linear-stub')).toBeFalsy()
        expect(wrapper.text()).not.toContain('visualization.progressing')
        expect(wrapper.text()).not.toContain('visualization.no_data')
      })
      it('renders with progress', () => {
        vis.data = undefined
        vis.progress = 0.74
        const wrapper = getWrapper(vis)
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.contains('v-progress-linear-stub')).toBeTruthy()
        expect(wrapper.text()).toContain('visualization.progressing')
        expect(wrapper.text()).not.toContain('visualization.no_data')
      })

      it('renders without data', () => {
        vis.data = undefined
        vis.progress = undefined
        const wrapper = getWrapper(vis)
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.contains('v-progress-linear-stub')).toBeFalsy()
        expect(wrapper.text()).not.toContain('visualization.progressing')
        expect(wrapper.text()).toContain('visualization.no_data')
      })
      it('renders without progress', () => {
        vis.data = {
          foo: 'bar'
        }
        vis.progress = undefined
        const wrapper = getWrapper(vis)
        expect(wrapper.html()).toMatchSnapshot()
        expect(wrapper.contains('v-progress-linear-stub')).toBeFalsy()
        expect(wrapper.text()).not.toContain('visualization.progressing')
        expect(wrapper.text()).not.toContain('visualization.no_data')
        // @ts-ignore
        expect(wrapper.vm.progress).toEqual(0)
      })
    })
  })

  it('renders the page if no visualization exists', () => {
    const wrapper = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the page for an invalid visualization', () => {
    const vis = new WordCloudVisualization({
      id: '1',
      name: 'My Visualization'
    })
    vis.key = 'invalid'
    const wrapper = getWrapper(vis)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
