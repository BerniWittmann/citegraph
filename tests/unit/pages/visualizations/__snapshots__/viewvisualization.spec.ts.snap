// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/visualizations/EditVisualization.vue renders the page for an invalid visualization 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.no_data</span></h3>
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for barchart renders 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub tag="div">
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for barchart renders with progress 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.progressing</span></h3>
        <v-progress-linear-stub color="#F37F21" value="74" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="25" rounded="true"><strong class="white--text">visualization.progress</strong></v-progress-linear-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for barchart renders without data 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.no_data</span></h3>
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for barchart renders without progress 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub tag="div">
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for network renders 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub tag="div">
        <network-stub visualization="[object Object]"></network-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for network renders with progress 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.progressing</span></h3>
        <v-progress-linear-stub color="#F37F21" value="74" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="25" rounded="true"><strong class="white--text">visualization.progress</strong></v-progress-linear-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for network renders without data 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.no_data</span></h3>
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for network renders without progress 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub tag="div">
        <network-stub visualization="[object Object]"></network-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for wordcloud renders 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub tag="div">
        <wordcloud-stub visualization="[object Object]"></wordcloud-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for wordcloud renders with progress 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.progressing</span></h3>
        <v-progress-linear-stub color="#F37F21" value="74" active="true" backgroundcolor="#FBBF35" backgroundopacity="0.5" buffervalue="100" height="25" rounded="true"><strong class="white--text">visualization.progress</strong></v-progress-linear-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for wordcloud renders without data 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub lg="3" tag="div" class="d-flex justify-center flex-column text-center">
        <empty-icon-stub class="align-self-center mb-6"></empty-icon-stub>
        <h3 class="headline mb-5"><span>visualization.no_data</span></h3>
        <!---->
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page for wordcloud renders without progress 1`] = `
<div class="visualization-view">
  <v-container-stub tag="div">
    <v-row-stub tag="div">
      <v-col-stub tag="div" class="d-flex align-center justify-space-between">
        <h1 class="display-1 mt-5">visualization.title</h1>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
          <v-icon-stub>mdi-pencil</v-icon-stub>
          visualization.edit_button
        </v-btn-stub>
      </v-col-stub>
    </v-row-stub>
    <v-row-stub tag="div" justify="center">
      <v-col-stub tag="div">
        <wordcloud-stub visualization="[object Object]"></wordcloud-stub>
      </v-col-stub>
    </v-row-stub>
  </v-container-stub>
</div>
`;

exports[`pages/visualizations/EditVisualization.vue renders the page if no visualization exists 1`] = `
<div class="visualization-view">
  <!---->
</div>
`;
